libnd4j/include/ops/declarable/generic/broadcastable/multiply.cpp

```diff
@@ -16,160 +16,158 @@
-                Nd4jLong *newshape;
-                COPY_SHAPE(x, newshape);
-
-                shapeList->push_back(newshape);
-            } else if (shape::isScalar(x) && !shape::isScalar(y)) {
-                Nd4jLong *newshape;
-                COPY_SHAPE(y, newshape);
-
-                shapeList->push_back(newshape);
-            } else if (!shape::isScalar(x) && shape::isScalar(y)) {
-                Nd4jLong *newshape;
-                COPY_SHAPE(x, newshape);
-
-                shapeList->push_back(newshape);
-            } else if (ShapeUtils<T>::areShapesBroadcastable(x, y)) {
-                Nd4jLong *newshape = nullptr;
-                ShapeUtils<T>::evalBroadcastShapeInfo(x, y, true, newshape, block.workspace());
-
-                shapeList->push_back(newshape);
-            } else {
-                // in this case we'll throw exception later
-                Nd4jLong *newshape;
-                COPY_SHAPE(x, newshape);
-
-                shapeList->push_back(newshape);
-            }
-
-            return shapeList;
-        }
-
-        CUSTOM_OP_IMPL(multiply_bp, 3, 2, false, 0, 0) {
-            auto x = INPUT_VARIABLE(0);
-            auto y = INPUT_VARIABLE(1);
-            auto epsNext = INPUT_VARIABLE(2);
-
-            auto gradX = OUTPUT_VARIABLE(0);
-            auto gradY = OUTPUT_VARIABLE(1);
-
-            auto lambdaX = LAMBDA_TT(_e, _y) {
-                return _e * _y;
-            };
-
-            auto lambdaY = LAMBDA_TT(_e, _x) {
-                return _e * _x;
-            };
-
-
-            if (x->isSameShape(y)) {
-                // PWT case case
-
-                // X gradient
-                epsNext->applyPairwiseLambda(y, lambdaX, gradX);
-
-                // Y gradient
-                epsNext->applyPairwiseLambda(x, lambdaY, gradY);
-
-            } else if (y->isScalar()) {
-                // scalar case
-                T _y = y->getScalar(0);
-                auto lambdaS = LAMBDA_T(_e, _y) {
-                    return _e * _y;
-                };
-
-                T tmpX = x->template reduceNumber<simdOps::Sum<T>>();
-                gradY->assign(tmpX);
-                
-                epsNext->applyLambda(lambdaS, gradX);
-            } else {
-                // broadcast case
+namespace ops {
+    
+CUSTOM_OP_IMPL(multiply, 2, 1, true, 0, 0) {
+    
+    NDArray<T> *x = INPUT_VARIABLE(0);
+    NDArray<T> *y = INPUT_VARIABLE(1);
+    NDArray<T> *z = OUTPUT_VARIABLE(0);
+
+    Nd4jLong* zShapeInfo = nullptr;
+    const bool areShapesBroadcastable = ShapeUtils<T>::evalBroadcastShapeInfo(x->getShapeInfo(), y->getShapeInfo(), true, zShapeInfo, block.getWorkspace());
+    REQUIRE_TRUE(areShapesBroadcastable, 0, "MULTIPLY OP: the shapes of x %s and y %s are not suitable for broadcast !", ShapeUtils<T>::shapeAsString(x).c_str(), ShapeUtils<T>::shapeAsString(y).c_str());
+    RELEASE(zShapeInfo, block.getWorkspace());
+
+    z->assign(*x * *y); 
```

Comment:
Please get back original FF implementation that utilizes BroadcastHelper. No need inventing the wheel once again.

Reply:
done